plugins/cicd-statistics/src/entity-page.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  ChartFilter,
+  ChartFilters,
+  getDefaultChartFilter,
+  getDefaultViewOptions,
+  ViewOptions,
+} from './components/chart-filters';
+import { AbortError, CicdConfiguration } from './apis';
+import { cleanupBuildTree } from './utils/stage-names';
+import { renderFallbacks, useAsyncChain } from './components/progress';
+
+export function EntityPageCicdCharts() {
+  const state = useCicdConfiguration();
+
+  return renderFallbacks(state, value => (
+    <CicdCharts cicdConfiguration={value} />
+  ));
+}
+
+const useStyles = makeStyles<Theme>(
+  theme => ({
+    pane: {
+      padding: theme.spacing(1, 1, 1, 1),
+    },
+  }),
+  {
+    name: 'CicdStatisticsView',
+  },
+);
+
+interface CicdChartsProps {
+  cicdConfiguration: CicdConfiguration;
+}
+
+function CicdCharts(props: CicdChartsProps) {
+  const { cicdConfiguration } = props;
+
+  const errorApi = useApi(errorApiRef);
+  const { entity } = useEntity();
+
+  const classes = useStyles();
+
+  const [chartFilter, setChartFilter] = useState(
+    getDefaultChartFilter(cicdConfiguration),
+  );
+  const [fetchedChartData, setFetchedChartData] = useState({
+    abortController: null as null | AbortController,
+    chartFilter,
+  });
+
+  const [viewOptions, setViewOptions] = useState(
+    getDefaultViewOptions(cicdConfiguration),
+  );
+
+  const fetchStatisticsOptions = useMemo((): UseCicdStatisticsOptions => {
+    const abortController = new AbortController();
+    fetchedChartData.abortController = abortController;
+    return {
+      abortController,
+      entity,
+      timeFrom: startOfDay(fetchedChartData.chartFilter.fromDate),
+      timeTo: endOfDay(fetchedChartData.chartFilter.toDate),
+      filterStatus: fetchedChartData.chartFilter.status,
+      filterType: fetchedChartData.chartFilter.branch,
+    };
+  }, [entity, fetchedChartData]);
+
+  const statisticsState = useCicdStatistics(fetchStatisticsOptions);
+
+  const updateFilter = useCallback(() => {
+    // Abort previous fetch
+    fetchedChartData.abortController?.abort();
+
+    setFetchedChartData({ abortController: null, chartFilter });
+  }, [fetchedChartData, setFetchedChartData, chartFilter]);
+
+  const chartableStagesState = useAsyncChain(
+    statisticsState,
+    async value =>
+      buildsToChartableStages(
+        await cleanupBuildTree(value.builds, {
+          formatStageName: cicdConfiguration.formatStageName,
+          lowerCase: viewOptions.lowercaseNames,
+        }),
+        { normalizeTimeRange: viewOptions.normalizeTimeRange },
+      ),
+    [statisticsState, cicdConfiguration, viewOptions],
+  );
+
+  const onFilterChange = useCallback((filter: ChartFilter) => {
+    setChartFilter(filter);
+  }, []);
+
+  const onViewOptionsChange = useCallback(
+    (options: ViewOptions) => {
+      setViewOptions(options);
+    },
+    [setViewOptions],
+  );
+
+  useEffect(() => {
+    if (
+      !chartableStagesState.error ||
+      chartableStagesState.error instanceof AbortError
```

Comment:
I'd recommend `error.name === 'AbortError'` as `instanceof` is pretty unreliable when you start mixing package versions. It also makes the checking a bit more lax so that you end up supporting the error thrown by `fetch(..., { signal })`

Reply:
Good point!